core/src/debt.ts

```diff
@@ -0,0 +1,239 @@
+};
+export const collectDebtAuction = async (network: string, auctionIndex: number, notifier?: Notifier) => {
+    const auction = await getActiveDebtAuctionOrUndefined(network, auctionIndex);
+    if (!auction || auction.state !== 'ready-for-collection') {
+        throw new Error('Did not find the auction to collect.');
+    }
+    await executeTransaction(network, 'MCD_FLOP', 'deal', [auctionIndex], { notifier });
+};
+export const enrichDebtAuction = async (
+    network: string,
+    auction: DebtAuctionActive
+): Promise<DebtAuctionTransaction> => {
+    const nextMaximumLotReceived = await getNextMaximumMkrReceived(network, auction);
+    const unitPrice = auction.bidAmountDai.div(auction.receiveAmountMKR);
+    const marketUnitPrice = await getMarketPriceDaiToMkr(network, auction.receiveAmountMKR);
+    const marketUnitPriceToUnitPriceRatio = unitPrice.minus(marketUnitPrice).dividedBy(marketUnitPrice);
+    const fees = await getDebtAuctionTransactionFees(network);
+    return {
+        ...auction,
+        ...fees,
+        nextMaximumLotReceived,
+        unitPrice,
+        marketUnitPrice,
+        marketUnitPriceToUnitPriceRatio,
+    };
+};
+
+export const enrichDebtAuctions = async (network: string, auctions: DebtAuction[]): Promise<DebtAuction[]> => {
+    const auctionsWithNextMinimumBidsPromises = auctions.map(auction => {
+        if (auction.state === 'collected') {
+            return auction;
+        }
+        return enrichDebtAuction(network, auction);
+    });
+    return await Promise.all(auctionsWithNextMinimumBidsPromises as Promise<DebtAuctionTransaction>[]);
+};
+
+export const getDebtAuctionLastIndex = async (contract: Contract): Promise<number> => {
+    const auctionsQuantityBinary = await contract.kicks();
+    return new BigNumber(auctionsQuantityBinary._hex).toNumber();
+};
+
+const _getDebtAuctionBidIncreaseCoefficient = async (
+    network: string,
+    contractName: 'MCD_FLAP' | 'MCD_FLOP'
+): Promise<BigNumber> => {
+    const contract = await getContract(network, contractName);
+    const auctionsQuantityBinary = await contract.beg();
+    return new BigNumber(auctionsQuantityBinary._hex).shiftedBy(-MKR_NUMBER_OF_DIGITS);
+};
+
+export const getDebtAuctionBidIncreaseCoefficient = memoizee(_getDebtAuctionBidIncreaseCoefficient, {
+    promise: true,
+    length: 3,
+});
+
+const getDebtAuctionState = async (network: string, earliestEndDate: Date, greatestBid: number) => {
+    const isBidExpired = (await getNetworkDate(network)) > earliestEndDate;
+    const haveBids = greatestBid !== 0;
+    if (haveBids) {
+        if (isBidExpired) {
+            return 'ready-for-collection';
+        }
+        return 'have-bids';
+    }
+    if (isBidExpired) {
+        return 'requires-restart';
+    }
+    return 'just-started';
+};
+
+const _getContractAuctionDuration = async (network: string, contractName: 'MCD_FLAP' | 'MCD_FLOP') => {
+    const contract = await getContract(network, contractName);
```

Comment:
Do you want to clean up unrelated logic and remove contract name parameter in all those functions?
```suggestion
    const contract = await getContract(network, CONTRACT);
```